bin/prep_ap_pipe.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# A script to set things up to run ap_pipe via slurm on some DECam data.
+#
+# First, run this script:
+#       $ bash prep_ap_pipe.sh $1=rerun $2=repo $3=calib $4=template $5=filter
+# e.g., $ bash prep_ap_pipe.sh slurm1 . calibdir templatedir g
+#
+# (Note each argument is optional and is set with defaults below)
+#
+# This will create two new files: CONFFILE and RUNFILE
+# (called run_ap_pipe.conf and run_ap_pipe.sh, respectively).
+#
+# Then, you are ready to submit a slurm job:
+#       $ sbatch run_ap_pipe.sl $1=repo $2=filter
+# e.g., $ sbatch run_ap_pipe.sl . g
+#
+# (Note you MUST input repo and filter to the sbatch call.)
+
+# Get directory of this shell script file
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+
+# Set environment variables used when running ap_pipe.py
+RERUN=${1:-slurm1}
+REPO=${2:-/project/mrawls/hits2015/}
+CALIB=${3:-/project/mrawls/hits2015/calib3}
+TEMPLATE=${4:-/project/mrawls/hits2015/templates}
+FILTERNAME=${5:-g}
+
+# The result of running this script is to create a .conf file
+# (Before we write to it, we remove any old one lying around)
+# Also create a directory to write slurm outfiles to if it doesn't exist yet
+CONFFILE=run_ap_pipe.conf
+rm -f ${CONFFILE}
+mkdir -p slurm
+
+# Create CONFFILE
+# Loop over DECam CCDs (note ccdnums 2 and 61 will fail)
+for ccdidx in {1..62};
```

Comment:
If you add a NCCDS argument above (defaulting to 62) this will then be generic to non-DECam cameras.

Reply:
Sadly not, because decam has the field "ccdnum" while other cameras use "ccd." (That's one reason I didn't bother doing this.)

Reply:
Sounds like it's time for another variable!

(This does make me think that doing this in stack-land may be easier eventually.  Or harmonizing our obs packages further...)